sklearn/metrics/_classification.py

```diff
@@ -2419,6 +2426,11 @@ def balanced_accuracy_score(y_true, y_pred, *, sample_weight=None, adjusted=Fals
         performance would score 0, while keeping perfect performance at a score
         of 1.
 
+    zero_division : {"warn", 0, 1}, default="warn"
```

Comment:
```suggestion
    zero_division : {"warn", 0.0, 1.0, np.nan}, default="warn"
```

Reply:
To be conistent with other metrics, we need to offer the `np.nan` option as well.